src/core/qgstemporalnavigationobject.cpp

```diff
@@ -128,7 +128,11 @@ void QgsTemporalNavigationObject::setNavigationMode( const NavigationMode mode )
 
 void QgsTemporalNavigationObject::setTemporalExtents( const QgsDateTimeRange &temporalExtents )
 {
-  mTemporalExtents = temporalExtents;
+  if ( !( mTemporalExtents == temporalExtents ) )
+  {
+    mTemporalExtents = temporalExtents;
+    emit temporalExtentsChanged( mTemporalExtents );
+  }
```

Comment:
```
if ( mTemporalExtents == temporalExtents )
    return;
mTemporalExtents = temporalExtents;
emit temporalExtentsChanged( mTemporalExtents );
```
I think if we check it using the above way, we will be able to skip the execution of the following switch block.

Reply:
Hi @Samweli, Thanks. Yes I was thinking about that too, but I was not sure how essential it was to call the switch that followed.. If you think this is OK, then I will do this.

Can we/you fix this here? Or should I just do a commit?

Reply:
Hi @rduivenvoorde, nice, I could, but I think you should just do a commit...